src/app/pages/wallets/my-wallets/pages-wallets-my-wallet.model.ts

```diff
@@ -1,13 +1,25 @@
-import { IWalletApiResponse } from '../../../domains/wallets/domains.wallets.types';
+import { IWalletApiResponse, IWalletPayload } from '../../../domains/wallets/domains.wallets.types';
 
 export class MyWallet {
-  public readonly id: number;
+  public readonly id: number | null;
   public readonly createdAt: Date;
   public readonly name: string;
 
   public constructor(data: IWalletApiResponse) {
-    this.id = data.id;
+    this.id = data.id === 0 ? null : data.id;
     this.createdAt = new Date(data.creationDate);
     this.name = data.name;
   }
+
+  public toPayload(): IWalletPayload {
+    return { name: this.name }
+  }
+
+  public static toAddNewWallet(name: string, creationDate: string = '2022-10-23T09:47:52.595721658Z'): MyWallet {
```

Comment:
Instead of those two static methods I would create one more generic / reusable called `public static create(data: Partial<IWalletApiResponse): MyWallet`. It should basically provide fallback for all properties that won't be provided so for example:
- if `id` won't be provided it should be null
- if `name` won't be provided it should be empty string
- if `creationDate` won't be provided it should be `new Date().toString()`